### PR TITLE
[model] fix: Match HF proportional RoPE shape in Gemma 4 global attention

### PR DESCRIPTION
**Pull request by jeje-soy-michu**
# What does this PR do ?

Fixes the global-attention rotary embedding in `Gemma4RotaryEmbedding` so its
`inv_freq` matches Hugging Face's proportional RoPE layout byte-for-byte. This
affects **all Gemma 4 variants** (MoE 26B-A4B, Dense 31B, and VL), since they all
build the global RoPE through `Gemma4ModelProvider.provide()`.

# Changelog

- `gemma4_provider.py`: pass `rotary_percent=1.0` to `RotaryEmbedding.__init__`
  (was `global_rotary_percent=0.25`) so the base builds machinery on the full `head_dim`.
- `gemma4_provider.py`: construct `inv_freq` as `cat(64 real freqs, zeros(head_dim//2 - 64))`
  so `rotate_half` pairs lanes `(i, i + head_dim//2)` exactly as HF does.
- `tests/unit_tests/models/gemma/test_gemma4_provider.py`: add
  `TestGemma4RotaryEmbeddingProportionalShape` asserting the global `inv_freq` length
  (`head_dim//2`), the real/zero-pad split (64 + 192), and the proportional-denominator
  values. Verified to fail on the pre-fix code.

# Scope

- Single shared code path: `Gemma4RotaryEmbedding` is instantiated only in
  `Gemma4ModelProvider.provide()`. All variants route through it:
  - MoE `gemma-4-26B-A4B`
  - Dense `gemma-4-31B`
  - Text / VL
- Only **global / `full_attention`** layers are affected; local sliding layers use a
  separate full-rotary `rope_local` (`rope_type="default"`) and are unchanged.

# Reference (HF official)

Reference implementation: `_compute_proportional_rope_parameters` in 🤗 transformers
`src/transformers/modeling_rope_utils.py`:

```python
rope_angles = int(rope_proportion * head_dim // 2)        # 64
inv_freq_rotated = 1.0 / (base ** (arange(0, ...) / head_dim))
nope_angles = head_dim // 2 - rope_angles                 # 192
inv_freq = cat([inv_freq_rotated, zeros(nope_angles)])    # length head_dim//2 = 256
```

Docstring: *"proportional RoPE will always return an encoding that is the size of
`head_dim`."* Mainline produced length 64 instead of 256.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)
in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve
and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [x] Did you write any new necessary tests? (inv_freq-shape regression test)
- [x] Did you add or update any necessary documentation? (n/a)
- [x] Does the PR affect components that are optional to install? (No)

# Additional Information

- Related to #4195

### CODE REVIEW

File: src/megatron/bridge/models/gemma/gemma4_provider.py [hunk 1]
```diff
@@ -653,17 +653,22 @@ def __init__(
         super().__init__(
             kv_channels=global_kv_channels,
             rotary_base=rotary_base,
-            rotary_percent=global_rotary_percent,
```
Comment (yaoyu-33) [inline src/megatron/bridge/models/gemma/gemma4_provider.py:656]: why is this change though?

Comment (jeje-soy-michu) [reply src/megatron/bridge/models/gemma/gemma4_provider.py:656]: `rotary_percent` is only used by the parent to compute the dim of inv_freq but the forward path derives the rotary width purely from `self.inv_freq`'s length. Since we overwrite `self.inv_freq` a few lines later the value passed here doesn't affect the final RoPE.

I set it to 1.0 (vs the previous 0.25) only so the parent's transient inv_freq is built at the full head_dim and stays consistent with the length we then assign. The actual fix is the torch.cat([real_freqs, zeros(...)]). Happy to add a clarifying comment, or revert this line to global_rotary_percent since it's functionally a no-op once inv_freq is overwritten.

**Comment by jeje-soy-michu** [comment]
No longer needed, the fix was already merged https://github.com/NVIDIA-NeMo/Megatron-Bridge/pull/4148
